internal/catalogd/garbagecollection/garbage_collector.go

```diff
@@ -79,7 +79,8 @@ func runGarbageCollection(ctx context.Context, cachePath string, metaClient meta
 	if err != nil {
 		return nil, fmt.Errorf("error reading cache directory: %w", err)
 	}
-	removed := []string{}
+	// Pre-allocate removed slice with estimated capacity to avoid reallocation
+	removed := make([]string, 0, len(cacheDirEntries))
```

Comment:
Looks like the loop below is appending to this slice still, that needs to be modified now right?